frontend/lib/resolution/VarScopeVisitor.h

```diff
@@ -140,6 +153,22 @@ class VarScopeVisitor {
       Only appropriate for expressions (not for Loops) */
   void handleMentions(const AstNode* ast, RV& rv);
 
+  /** Handle updating initedVars if an assignment represents a split-init.
+      Returns true if it was a split init. */
+  bool handleSplitInitAssign(const OpCall* ast,
+                             const std::set<ID>& allSplitInitedVars,
+                             RV& rv);
+
+  /** Handle updating initedVars if a call with at 'out' formal
+      represents a split-init. Returns true if it was a split init. */
+  bool handleSplitInitOut(const FnCall* ast,
+                          const AstNode* actual,
+                          const std::set<ID>& allSplitInitedVars,
+                          RV& rv);
```

Comment:
Argh.

I think I understand why these are methods on the VarScopeVisitor; the two subsequent "passes" (copy elision and initialization-deinitialization) require information about the first pass (split init). Since the logic in the two child classes is identical, it makes sense to lift it to a parent class.

On the other hand, it's strange for the _parent_ to depend on the state of its children (allSplitInitedVars is a field on both init-deinit and elision, right?). In a pure OOP world, the "ideal" solution would be yet another parent class sitting in the hierarchy between VarSopeVisitor and CallInitDeinit, which contains as fields the `allSplitInitedVars` set, and provides `handleSplitInitOut`. I think it might be messy / verbose to introduce this parent class, though. What do you think?

Reply:
IMO, it's not a problem to pass `allSplitInitedVars` from the child classes in the case this is needed. It seems clear enough and there is not much code duplication. I guess that doesn't bother me at all (so I wouldn't have the appetite for creating another class in the hierarchy because that complicates things).